### PR TITLE
Support Load Balancers for Digital Ocean

### PR DESCRIPTION
**Pull request by stgleb**

### CODE REVIEW

File: pkg/workflows/steps/digitalocean/create_load_balancer.go [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+func (s *CreateLoadBalancerStep) Run(ctx context.Context, output io.Writer, config *steps.Config) error {
+	lbSvc := s.getServices(config.DigitalOceanConfig.AccessToken)
+
+	req := &godo.LoadBalancerRequest{
+		Name:   util.CreateLBName(config.ClusterID, true),
+		Region: config.DigitalOceanConfig.Region,
+		ForwardingRules: []godo.ForwardingRule{
+			{
+				EntryPort:      443,
```
Comment (saheienko) [inline pkg/workflows/steps/digitalocean/create_load_balancer.go]: Could this be set as a configuration parameter?

File: pkg/workflows/steps/digitalocean/create_load_balancer.go [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+			ResponseTimeoutSeconds: 10,
+		},
+		Tag: fmt.Sprintf("master-%s", config.ClusterID),
+	}
+
+	externalLoadBalancer, _, err := lbSvc.Create(ctx, req)
+
+	if err != nil {
+		logrus.Errorf("Error while creating external load balancer %v", err)
+		return errors.Wrapf(err, "Error while creating external load balancer")
```
Comment (saheienko) [inline pkg/workflows/steps/digitalocean/create_load_balancer.go]: What the purpose of logging and returning an error?

File: pkg/workflows/steps/digitalocean/create_load_balancer.go (hunk 1 above)
Comment (stgleb) [reply pkg/workflows/steps/digitalocean/create_load_balancer.go]: Will be done

File: pkg/workflows/steps/digitalocean/create_load_balancer.go (hunk 2 above)
Comment (stgleb) [reply pkg/workflows/steps/digitalocean/create_load_balancer.go]: Well, I've done it to be follow style of other steps.

File: pkg/workflows/steps/digitalocean/create_load_balancer.go (hunk 1 above)
Comment (stgleb) [reply pkg/workflows/steps/digitalocean/create_load_balancer.go]: I am planning to extract it to separate argument on the UI, also to store it in Kube, but I would like to do that with separate PR.

File: pkg/clouds/constants.go [hunk 3]
```diff
@@ -60,6 +62,8 @@ const (
 	AwsMasterInstanceProfile    = "aws_master_instance_profile"
 	AwsNodeInstanceProfile      = "aws_node_instance_profile"
 	AwsImageID                  = "aws_image_id"
+	AwsExternalLoadBalancerName = "AwsExternalLoadBalancerName"
+	AwsInternalLoadBalancerName = "AwsInternalLoadBalancerName"
```
[the remark this answers is not in the document]
Comment (stgleb) [reply pkg/clouds/constants.go]: Will be done

File: pkg/workflows/steps/digitalocean/create_load_balancer.go [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+				TargetProtocol: "HTTPS",
+				// NOTE(stgleb): Sticky sessions won't work with TLS passthrough
+				// https://www.digitalocean.com/docs/networking/load-balancers/how-to/ssl-passthrough/
+				TlsPassthrough: true,
+			},
+		},
+		HealthCheck: &godo.HealthCheck{
+			Protocol:               "TCP",
+			Port:                   443,
+			CheckIntervalSeconds:   10,
```
Comment (taaraora) [inline pkg/workflows/steps/digitalocean/create_load_balancer.go]: Maybe we need to have a config for all these numbers?

File: pkg/workflows/steps/digitalocean/create_load_balancer.go [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+			HealthyThreshold:       3,
+			ResponseTimeoutSeconds: 10,
+		},
+		Tag: fmt.Sprintf("master-%s", config.ClusterID),
+	}
+
+	externalLoadBalancer, _, err := lbSvc.Create(ctx, req)
+
+	if err != nil {
+		logrus.Errorf("Error while creating external load balancer %v", err)
```
Comment (taaraora) [inline pkg/workflows/steps/digitalocean/create_load_balancer.go]: why do we log the same error which is returning on the upper layer and is logged there?

File: pkg/workflows/steps/digitalocean/create_load_balancer.go [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+		if err == nil && externalLoadBalancer.Status == StatusActive {
+			break
+		}
+
+		time.Sleep(timeout)
+		timeout = timeout * 2
+	}
+
+	if err != nil {
+		logrus.Errorf("Error while getting external load balancer %v", err)
```
Comment (taaraora) [inline pkg/workflows/steps/digitalocean/create_load_balancer.go]: excessive logging

File: pkg/workflows/steps/digitalocean/create_load_balancer.go (hunk 4 above)
Comment (stgleb) [reply pkg/workflows/steps/digitalocean/create_load_balancer.go]: In that case config for user on UI  will be unreasonable big

File: pkg/workflows/steps/digitalocean/create_load_balancer.go (hunk 5 above)
Comment (stgleb) [reply pkg/workflows/steps/digitalocean/create_load_balancer.go]: I agree, but I've just followed SG style

**Pull request merged**